lib/internal/http2/core.js

```diff
@@ -1912,10 +1912,19 @@ class Http2Stream extends Duplex {
 
     let req;
 
+    // writeGeneric does not destroy on error and we cannot enable autoDestroy,
+    // so make sure to destroy on error.
+    const callback = err => {
+      if (err) {
+        this.destroy(err);
+      }
+      cb(err);
+    }
```

Comment:
@addaleax sorry... :D

Reply:
@ronag Do you know why we would want to do this for http2 but not for net?

Reply:
@addaleax: This PR enables `autoDestroy` for `net`. But not for `http2` (too hard for me :/).

Reply:
I promise to try and dig into the `http2` part. However, that is significantly more complicated and will make this much harder to land.

Reply:
Yeah, that’s fine, although we definitely should aim to have `autoDestroy` for http2 as well…

Reply:
@addaleax: Totally agree. I did spend sometime on that but once I realised how significant changes that requires I kind of put it on hold for now...